napari/layers/base/base.py

```diff
@@ -951,45 +877,41 @@ def _slice_dims(self, point=None, ndisplay=2, order=None):
             rendered in canvas.
         """
         if point is None:
-            ndim = self.ndim
+            point = (0,) * self.ndim
         else:
-            ndim = len(point)
+            point = tuple(point)
+
+        ndim = len(point)
+        assert (
+            ndim >= self.ndim
+        ), 'slicing with fewer dimensions than layer has'
```

Comment:
Is the use of assert here intended for performance improvement from debug to production? If not, we should use normal `raise` syntax to avoid missing the error.

Reply:
I mostly added it to quickly check if that case was ever being hit mostly to check if we ever hit in the `offset < 0` case in `_world_to_data_dims_displayed`. I'll probably just remove it for now, as I think we should be able to progress in that case (as we do on `main` now).